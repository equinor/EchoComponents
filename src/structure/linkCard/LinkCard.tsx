import { Typography } from '@equinor/eds-core-react';
import React, { CSSProperties } from 'react';
import styles from './linkCard.module.css';

export interface LinkCardProps {
    image: string;
    title: string;
    subTitle: string;
    links: LinkCardItem[];
    style: CSSProperties;
}

export interface LinkCardItem {
    title: string;
    url: string;
    target?: '_blank' | '_self';
}

export const LinkCard: React.FC<LinkCardProps> = ({
    style,
    image,
    title,
    subTitle,
    links
}: LinkCardProps): JSX.Element => {
    return (
        <div className={styles.wrapper} style={style}>
            <div
                className={styles.image}
                style={{
                    backgroundImage: `url(${image})`
                }}
            ></div>
            <div className={styles.header}>
                <Typography className={styles.title} variant="h3">
                    {title}
                </Typography>
                <Typography className={styles.subTitle} variant="h3">
                    {subTitle}
                </Typography>
            </div>
            <div className={styles.links}>
                {links.map((link, index) => {
                    return (
                        <a href={link.url} key={index} target={link.target ?? '_self'}>
                            <Typography className={styles.link} variant="body_short">
                                {link.title}
                            </Typography>
                        </a>
                    );
                })}
            </div>
        </div>
    );
};
