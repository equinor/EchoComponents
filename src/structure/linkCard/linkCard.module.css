.wrapper {
    width: 310px;
    height: 582px;
    display: flex;
    flex-direction: column;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
    border-radius: 4px;
    background-color: var(--white);
    border: 1px solid var(--default);
    overflow: hidden;
}

.image {
    width: 100%;
    height: 247px;
    background-size: cover;
    background-position-x: center;
    background-position-y: center;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
}

.header {
    border-bottom: 1px solid var(--default);
    margin-right: 20px;
    margin-left: 20px;
    padding-top: 15px;
    padding-bottom: 15px;
    height: 96px;
}

.title {
    color: var(--equiGreen1) !important;
    margin: 0 0 15px 0 !important;
    line-height: 1em !important;
}

.subTitle {
    color: var(--black) !important;
    line-height: 1em !important;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
}

.links {
    padding-left: 20px;
    padding-right: 20px;
}

.links a:last-child p {
    border-bottom: none;
}

.links a {
    text-decoration: none;
}

.links p {
    line-height: 23px;
    border-bottom: 1px solid var(--default);
    color: var(--defaultSeverityText);
    padding: 16px;
    font-size: 16px;
}

.link:hover {
    background-color: var(--searchBarBackground);
}

.link:active {
    background-color: var(--equiGreen3);
}
