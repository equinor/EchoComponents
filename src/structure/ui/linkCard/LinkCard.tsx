import { Typography } from '@equinor/eds-core-react';
import React, { CSSProperties } from 'react';
import styles from './linkCard.module.css';

export interface LinkCardProps {
    image: string;
    title: string;
    subTitle: string;
    links: LinkCardItem[];
    style: CSSProperties;
}

export interface LinkCardItem {
    title: string;
    url: string;
}

export const LinkCard: React.FC<LinkCardProps> = ({
    style,
    image,
    title,
    subTitle,
    links
}: LinkCardProps): JSX.Element => {
    return (
        <div className={styles.linkCard} style={style}>
            <div
                className={styles.image}
                style={{
                    backgroundImage: `url(${image})`
                }}
            ></div>
            <div className={styles.header}>
                <Typography variant="h3">{title}</Typography>
                <Typography variant="h3">{subTitle}</Typography>
            </div>
            <div className={styles.links}>
                {links.map((link, index) => {
                    return (
                        <a href={link.url} key={index}>
                            <div className={styles.link}>
                                <Typography variant="body_short">{link.title}</Typography>
                            </div>
                        </a>
                    );
                })}
            </div>
        </div>
    );
};
