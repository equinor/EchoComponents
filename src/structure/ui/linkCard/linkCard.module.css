.linkCard {
    min-height: 535px;
    display: flex;
    flex-direction: column;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
    border-radius: 4px;
    background-color: white;
    border: 1px solid #dcdcdc;
    overflow: hidden;
}

.image {
    width: 100%;
    height: 247px;
    background-size: cover;
    background-position-x: center;
    background-position-y: center;
}

.header {
    border-bottom: 1px solid #dcdcdc;
    padding: 15px;
}

.header h3 {
    line-height: 1em;
}

.header h3:nth-child(1) {
    color: var(--equiGreen1);
    margin: 0 0 5px 0;
}

.header h3:nth-child(2) {
    color: var(--black);
}

.links {
}

.links a {
    text-decoration: none;
}

.link {
    border-bottom: 1px solid #dcdcdc;
    color: #3d3d3d;
    padding: 15px;
}

.link p {
    font-size: 16px;
}
