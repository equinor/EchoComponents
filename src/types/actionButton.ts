export interface ActionButton {
    title: string;
    onClick: () => void;
    variant: 'contained' | 'outlined' | 'ghost' | 'ghost_icon' | undefined;
}
