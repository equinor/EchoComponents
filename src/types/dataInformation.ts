export interface DataInformation {
    numberOfItems: number;
    label: string;
    ariaLabel?: string;
    onTagInfoClicked: () => void;
    itemType: string;
    itemButtonClassName?: string;
}
