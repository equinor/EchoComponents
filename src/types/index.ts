export * from './actionButton';
export * from './dataInformation';
export * from './draggableItem';
export * from './expandableRowProps';
export * from './iconItem';
export * from './iconListItem';
export * from './linkCardItem';
