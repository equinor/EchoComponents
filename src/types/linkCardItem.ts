export interface LinkCardItem {
    title: string;
    url: string;
    target?: '_blank' | '_self';
}
