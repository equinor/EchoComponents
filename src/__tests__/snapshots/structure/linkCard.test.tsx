import React from 'react';
import renderer from 'react-test-renderer';
import { LinkCard } from '../../../structure/linkCard/LinkCard';
import { LinkCardItem } from '../../../types/linkCardItem';

it('renders LinkCardItem correctly', () => {
    const links: LinkCardItem[] = [
        {
            title: 'Full',
            url: '#'
        },
        {
            title: 'HVDC',
            url: 'https://yahoo.com',
            target: '_blank'
        },
        {
            title: 'MSF',
            url: '#'
        },
        {
            title: 'UPM',
            url: '#'
        }
    ];
    const tree = renderer
        .create(
            <LinkCard
                title={'Johan Sverdrup'}
                subTitle={'Processing Platform 2'}
                links={links}
                image={'https://assets.bwbx.io/images/users/iqjWHBFdfxIU/iRLqgF0OqRHg/v0/1200x900.jpg'}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
