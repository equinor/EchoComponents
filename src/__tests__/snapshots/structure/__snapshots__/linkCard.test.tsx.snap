// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LinkCardItem correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(https://assets.bwbx.io/images/users/iqjWHBFdfxIU/iRLqgF0OqRHg/v0/1200x900.jpg)",
      }
    }
  />
  <div
    className="header"
  >
    <h3
      className="sc-eCssSg hvXwcX title"
    >
      Johan Sverdrup
    </h3>
    <h3
      className="sc-eCssSg hvXwcX subTitle"
    >
      Processing Platform 2
    </h3>
  </div>
  <div
    className="links"
  >
    <a
      href="#"
      target="_self"
    >
      <p
        className="sc-eCssSg bHjGEK link"
      >
        Full
      </p>
    </a>
    <a
      href="https://yahoo.com"
      target="_blank"
    >
      <p
        className="sc-eCssSg bHjGEK link"
      >
        HVDC
      </p>
    </a>
    <a
      href="#"
      target="_self"
    >
      <p
        className="sc-eCssSg bHjGEK link"
      >
        MSF
      </p>
    </a>
    <a
      href="#"
      target="_self"
    >
      <p
        className="sc-eCssSg bHjGEK link"
      >
        UPM
      </p>
    </a>
  </div>
</div>
`;
